Tests/TestArgumentsHistory/FixtureArgumentsHistory.swift

```diff
@@ -0,0 +1,382 @@
+        fooFuncVal1StringArgValues.append(val1)
+
+        if let fooFuncVal1StringHandler = fooFuncVal1StringHandler {
+            fooFuncVal1StringHandler(val1)
+        }
+    }
+
+    var fooFuncVal2CallCount = 0
+    var fooFuncVal2ArgValues = [Int]()
+    var fooFuncVal2Handler: ((Int) -> ())?
+    func fooFunc(val2: Int)  {
+        fooFuncVal2CallCount += 1
+        fooFuncVal2ArgValues.append(val2)
+
+        if let fooFuncVal2Handler = fooFuncVal2Handler {
+            fooFuncVal2Handler(val2)
+        }
+    }
+}
+"""
+
+let argumentsHistoryGenericsCase = """
+/// \(String.mockAnnotation)
+protocol Foo {
+    func fooFunc<T: StringProtocol>(val1: T, val2: T?)
+    func barFunc<T: Sequence, U: Collection>(val: T) -> U
+}
+"""
+
+let argumentsHistoryGenericsCaseMock = """
+class FooMock: Foo {
+    init() { }
+
+    var fooFuncCallCount = 0
+    var fooFuncArgValues = [(Any, Any?)]()
+    var fooFuncHandler: ((Any, Any?) -> ())?
+    func fooFunc<T: StringProtocol>(val1: T, val2: T?) {
+        fooFuncCallCount += 1
+        fooFuncArgValues.append((val1, val2))
+
+        if let fooFuncHandler = fooFuncHandler {
+            fooFuncHandler(val1, val2)
+        }
+    }
+
+    var barFuncCallCount = 0
+    var barFuncArgValues = [Any]()
+    var barFuncHandler: ((Any) -> (Any))?
+    func barFunc<T: Sequence, U: Collection>(val: T) -> U {
+        barFuncCallCount += 1
+        barFuncArgValues.append(val)
+
+        if let barFuncHandler = barFuncHandler {
+            return barFuncHandler(val) as! U
+        }
+        fatalError("barFuncHandler returns can't have a default value thus its handler must be set")
+    }
+}
+"""
+
+let argumentsHistoryInoutCase = """
+/// \(String.mockAnnotation)
+protocol Foo {
+    func fooFunc(val: inout Int)
+    func barFunc(into val: inout Int)
+}
+"""
+
+let argumentsHistoryInoutCaseMock = """
+class FooMock: Foo {
+    init() { }
+
+    var fooFuncCallCount = 0
+    var fooFuncArgValues = [Int]()
+    var fooFuncHandler: ((inout Int) -> ())?
+    func fooFunc(val: inout Int) {
+        fooFuncCallCount += 1
+        fooFuncArgValues.append(val)
+
+        if let fooFuncHandler = fooFuncHandler {
+            fooFuncHandler(&val)
+        }
+    }
+
+    var barFuncCallCount = 0
+    var barFuncArgValues = [Int]()
+    var barFuncHandler: ((inout Int) -> ())?
+    func barFunc(into val: inout Int) {
+        barFuncCallCount += 1
+        barFuncArgValues.append(val)
+
+        if let barFuncHandler = barFuncHandler {
+        barFuncHandler(&val)
+        }
+    }
+}
+"""
+
+let argumentsHistoryHandlerCase = """
+/// \(String.mockAnnotation)
+protocol Foo {
+    func fooFunc(handler: () -> Int)
+    func barFunc(val: Int, handler: (String) -> Void)
+}
+"""
+
+let argumentsHistoryHandlerCaseMock = """
+class FooMock: Foo {
+    init() { }
+
+    var fooFuncCallCount = 0
+    var fooFuncHandler: ((() -> Int) -> ())?
+    func fooFunc(handler: () -> Int) {
+        fooFuncCallCount += 1
+
```

Comment:
ditto^

Reply:
Same as above.
`fooFunc(handler: () -> Int)` doesn't have any capturable arguments.

Reply:
ah, overlooked. this looks good then.